Q/qr_mumps/build_tarballs.jl

```diff
@@ -58,12 +58,12 @@ products = [
 dependencies = [
     Dependency(PackageSpec(name="OpenBLAS32_jll", uuid="656ef2d0-ae68-5445-9ca0-591084a874a2")),
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
-    Dependency(PackageSpec(name="SuiteSparse32_jll", uuid="ca45d3f4-326b-53b0-9957-23b75aacb3f2"); compat="5.10.1"),
+    Dependency(PackageSpec(name="SuiteSparse_jll", uuid="bea87d4a-7f5b-5778-9afe-8cc45184846c")),
     Dependency(PackageSpec(name="METIS_jll", uuid="d00139f3-1899-568f-a2f0-47f597d42d70")),
     Dependency(PackageSpec(name="SCOTCH_jll", uuid="a8d0f55d-b80e-548d-aff6-1a04c175f0f9"); compat="6.1.3")
     # Dependency(PackageSpec(name="libblastrampoline_jll", uuid="8e850b90-86db-534c-a0d3-1478176c7d93")) for Julia ≥ v"1.9"
     # Dependency(PackageSpec(name="StarPU_jll", uuid="e3ad0b27-b140-5312-a56e-059adfc55eb4"))
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"9.1.0", clang_use_lld=false)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.10", preferred_gcc_version=v"9.1.0", clang_use_lld=false)
```

Comment:
You can't change compat bounds without changing version number of a package, this PR should be reverted

Reply:
Because with SuiteSparse 7.7.0 they changed the soname of `libamd.so`, it's not anymore `libamd.so.2` but `libamd.so.3` now.
Because this version of SuiteSparse is only available on Julia 1.10 and we can't dynamically changed SuiteSparse_jll.Jl, I bumped the version number of `qr_mumps_jll.jl`.

Reply:
https://github.com/JuliaSmoothOptimizers/QRMumps.jl/issues/86

Reply:
But you didn't specify the version of SuiteSparse, v7.7 is not what's used in Julia v1.10

Reply:
Because I wasn't certain which version of SuiteSparse introduced the new SONAME, and also because not all SONAMEs of SuiteSparse libraries are updated when Tim Davis increments the major version number.

However, recompiling qr_mumps with SuiteSparse 7.7.0 resolved the issue.
So the SONAME probably changed with SuiteSparse 7.x.x.
I added a compatibility entry in Julia to install it only on Julia versions >= 1.10.